pkg/controller/namespace/namespace_controller.go

```diff
@@ -209,25 +209,61 @@ func deleteAllContent(kubeClient client.Interface, experimentalMode bool, namesp
 	return estimate, nil
 }
 
-// syncNamespace makes namespace life-cycle decisions
-func syncNamespace(kubeClient client.Interface, experimentalMode bool, namespace api.Namespace) (err error) {
-	if namespace.DeletionTimestamp == nil {
-		return nil
+// updateNamespaceFunc is a function that makes an update to a namespace
+type updateNamespaceFunc func(kubeClient client.Interface, namespace *api.Namespace) (*api.Namespace, error)
+
+// retryOnConflictError retries the specified delegate if there was a conflict error
+func retryOnConflictError(kubeClient client.Interface, namespace *api.Namespace, delegate updateNamespaceFunc) (result *api.Namespace, err error) {
```

Comment:
This isn't your fault, but needs to be in generic client.  Add a TODO that this should be a common behavior across clients.